core/lib/zksync_core/src/commitment_generator/metrics.rs

```diff
@@ -0,0 +1,29 @@
+use std::time::Duration;
+
+use vise::{Buckets, EncodeLabelSet, EncodeLabelValue, Family, Histogram, Metrics};
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, EncodeLabelValue, EncodeLabelSet)]
+#[metrics(label = "stage", rename_all = "snake_case")]
+pub(super) enum CommitmentStage {
+    PrepareInput,
+    Calculate,
+    SaveResults,
+}
+
+/// Metrics for the commitment generator.
+#[derive(Debug, Metrics)]
+#[metrics(prefix = "server_commitment_generator")]
+pub(super) struct CommitmentGeneratorMetrics {
+    /// Latency of generating commitment per stage.
+    #[metrics(buckets = Buckets::LATENCIES)]
```

Comment:
Nit: It's better to specify unit of measurement for new metrics: `unit = Unit::Seconds` (here, `Unit` should be imported into the scope from `vise`). We don't add UoM for the existing metrics because it influences metric name (as per the Open Metrics spec).